Index: README.md
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/README.md b/README.md
--- a/README.md	(revision 4354e09391062d73397bac1d3fb7c234b043694b)
+++ b/README.md	(revision 305a7c87d68c4096cf831406319efd477f59bab5)
@@ -9,7 +9,8 @@
 1. Download this module
 2. [Download DropzoneJS](https://github.com/enyo/dropzone) and place it in the
    libraries folder
-3. Install dropzonejs the [usual way](https://www.drupal.org/docs/extending-drupal/installing-drupal-modules)
+3. Install dropzonejs the [usual way](https://www.drupal.org/docs/extending-drupal/installing-drupal-modules).
+   Note: version 5.7.0+ of the library is required when using chunked uploads.
 4. Remove "test" folder from libraries folder as it could constitute a
    security risk to your site. See http://drupal.org/node/1189632 for more info.
 
Index: dropzonejs.libraries.yml
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/dropzonejs.libraries.yml b/dropzonejs.libraries.yml
--- a/dropzonejs.libraries.yml	(revision 4354e09391062d73397bac1d3fb7c234b043694b)
+++ b/dropzonejs.libraries.yml	(revision 305a7c87d68c4096cf831406319efd477f59bab5)
@@ -23,6 +23,7 @@
   dependencies:
     - core/jquery.once
     - core/drupal
+    - core/drupal.ajax
     - core/drupalSettings
     - core/underscore
     - dropzonejs/dropzonejs
Index: dropzonejs.routing.yml
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/dropzonejs.routing.yml b/dropzonejs.routing.yml
--- a/dropzonejs.routing.yml	(revision 4354e09391062d73397bac1d3fb7c234b043694b)
+++ b/dropzonejs.routing.yml	(revision 305a7c87d68c4096cf831406319efd477f59bab5)
@@ -5,3 +5,17 @@
   requirements:
     _permission: 'dropzone upload files'
     _csrf_token: 'TRUE'
+dropzonejs.chunked.upload:
+  path: '/dropzonejs/chunked/upload'
+  defaults:
+    _controller: '\Drupal\dropzonejs\Controller\ChunkedUploadController::handleUploads'
+  requirements:
+    _permission: 'dropzone upload files'
+    _csrf_token: 'TRUE'
+dropzonejs.chunked.finalize:
+  path: '/dropzonejs/chunked/finalize'
+  defaults:
+    _controller: '\Drupal\dropzonejs\Controller\ChunkedUploadController::finalizeUpload'
+  requirements:
+    _permission: 'dropzone upload files'
+    _csrf_token: 'TRUE'
Index: dropzonejs.services.yml
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/dropzonejs.services.yml b/dropzonejs.services.yml
--- a/dropzonejs.services.yml	(revision 4354e09391062d73397bac1d3fb7c234b043694b)
+++ b/dropzonejs.services.yml	(revision 305a7c87d68c4096cf831406319efd477f59bab5)
@@ -5,3 +5,6 @@
   dropzonejs.upload_handler:
     class: Drupal\dropzonejs\UploadHandler
     arguments: ['@request_stack', '@config.factory', '@transliteration', '@language_manager']
+  dropzonejs.chunked_upload_handler:
+    class: Drupal\dropzonejs\ChunkedUploadHandler
+    arguments: ['@request_stack', '@config.factory', '@transliteration', '@language_manager', '@file_system']
Index: js/dropzone.integration.js
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/js/dropzone.integration.js b/js/dropzone.integration.js
--- a/js/dropzone.integration.js	(revision 4354e09391062d73397bac1d3fb7c234b043694b)
+++ b/js/dropzone.integration.js	(revision 305a7c87d68c4096cf831406319efd477f59bab5)
@@ -8,7 +8,42 @@
 (function ($, Drupal, drupalSettings) {
   'use strict';
 
-  Drupal.dropzonejsInstances = [];
+  Drupal.dropzonejs = Drupal.dropzonejs || {};
+
+  Drupal.dropzonejs.instances = {};
+
+  Drupal.dropzonejs.chunksUploaded = function (file, done, el) {
+    var selector = $(el);
+    var input = selector.siblings('input');
+    var url = input.attr('data-finalize-path');
+
+    // Send request to merge chunks for current file.
+    Drupal.ajax({
+      url: url,
+      base: false,
+      element: false,
+      progress: false,
+      submit: {
+        filename: file.upload.filename,
+        dzuuid: file.upload.uuid,
+        dztotalfilesize: file.size,
+        dztotalchunkcount: file.upload.totalChunkCount
+      },
+      success: function (response) {
+        // Call dropzone's done callback. (see https://www.dropzonejs.com/#config-chunksUploaded)
+        // The file is transliterated on upload. The element has to reflect
+        // the real filename.
+        file.processedName = response.result;
+
+        done();
+      },
+      error: function (xhr, url) {
+        var instance = Drupal.dropzonejs.instances[selector.attr('id')];
+        var response = $.parseJSON(xhr.responseText);
+        instance._errorProcessing([file], response.error, xhr);
+      }
+    }).execute();
+  };
 
   /* global Dropzone */
   Drupal.behaviors.dropzonejsIntegraion = {
@@ -46,6 +81,23 @@
         };
         var instanceConfig = drupalSettings.dropzonejs.instances[selector.attr('id')];
 
+        if (instanceConfig.chunking) {
+          var el = this;
+          config.chunksUploaded = function(file, done) {
+            Drupal.dropzonejs.chunksUploaded(file, done, el);
+          };
+        }
+
+        // Mininum resolution check, based on example from https://gitlab.com/meno/dropzone/-/wikis/FAQ#reject-images-based-on-image-dimensions
+        var minResConfig = {};
+        if (instanceConfig.min_resolution_x) {
+          minResConfig = {
+            accept: function(file, done) {
+              file.acceptDimensions = done;
+              file.rejectDimensions = function(dimensions) { done(Drupal.t("Invalid dimension. Minimum dimension: ")+dimensions);};
+            }
+          }
+        }
         // If DropzoneJS instance is already registered on Element. There is no
         // need to register it again.
         if (selector.once('register-dropzonejs').length !== selector.length) {
@@ -77,11 +129,15 @@
           var uploadedFilesElement = selector.siblings(':hidden');
           var currentValue = uploadedFilesElement.attr('value') || '';
 
-          // The file is transliterated on upload. The element has to reflect
-          // the real filename.
-          file.processedName = response.result;
+          // Chunked uplaods get an empty response.
+          // (see https://gitlab.com/meno/dropzone/-/blob/master/src/dropzone.js#L1978).
+          if (response) {
+            // The file is transliterated on upload. The element has to reflect
+            // the real filename.
+            file.processedName = response.result;
+          }
 
-          uploadedFilesElement.attr('value', currentValue + response.result + ';');
+          uploadedFilesElement.attr('value', currentValue + file.processedName + ';');
         });
 
         // React on file removing.
@@ -111,6 +167,30 @@
             dropzoneInstance.removeFile(rejectedFiles[i]);
           }
         });
+// Rebuild Drupal core's minimum image resolution validation, as this
+        // does not get called in dropzone entity browser widgets
+        dropzoneInstance.on("thumbnail", function (file) {
+          var dimensions;
+          if (instanceConfig.min_resolution_x) {
+            dimensions = Drupal.t('width')+': '+instanceConfig.min_resolution_x+"px";
+            if (instanceConfig.min_resolution_y) {
+              dimensions = dimensions + ", "
+            }
+          }
+          if (instanceConfig.min_resolution_y) {
+            dimensions = dimensions + Drupal.t('height')+': '+instanceConfig.min_resolution_y+"px";
+          }
+          if ((instanceConfig.min_resolution_x && file.width < instanceConfig.min_resolution_x) || (instanceConfig.min_resolution_y) && file.height < instanceConfig.min_resolution_y) {
+            file.rejectDimensions(dimensions);
+          }
+          else {
+            if (typeof file.acceptDimensions !== "undefined") {
+              file.acceptDimensions();
+            }
+          }
+        });
+
+        Drupal.dropzonejs.instances[selector.attr('id')] = dropzoneInstance;
       });
     }
   };
Index: modules/eb_widget/config/schema/dropzonejs_eb_widget.schema.yml
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/modules/eb_widget/config/schema/dropzonejs_eb_widget.schema.yml b/modules/eb_widget/config/schema/dropzonejs_eb_widget.schema.yml
--- a/modules/eb_widget/config/schema/dropzonejs_eb_widget.schema.yml	(revision 4354e09391062d73397bac1d3fb7c234b043694b)
+++ b/modules/eb_widget/config/schema/dropzonejs_eb_widget.schema.yml	(revision 305a7c87d68c4096cf831406319efd477f59bab5)
@@ -38,6 +38,27 @@
     thumbnail_method:
       type: string
       label: 'Method used for generating the thumbnail'
+    min_resolution:
+      type: string
+      label: 'Minimum image resolution'
+    chunking:
+      type: boolean
+      label: 'Enable chunked uploads'
+    force_chunking:
+      type: boolean
+      label: 'Force each upload to be chunked, even if the file size is below chunk_size'
+    chunk_size:
+      type: integer
+      label: 'The chunk size in bytes'
+    parallel_chunk_uploads:
+      type: boolean
+      label: 'Upload multiple chunks in parallel'
+    retry_chunks:
+      type: boolean
+      label: 'Whether a chunk upload should be retried if it fails'
+    retry_chunks_limit:
+      type: integer
+      label: 'Number of times a chunk upload should be retried'
 
 entity_browser.browser.widget.dropzonejs_media_entity:
   type: entity_browser.browser.widget.dropzonejs
Index: modules/eb_widget/dropzonejs_eb_widget.install
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/modules/eb_widget/dropzonejs_eb_widget.install b/modules/eb_widget/dropzonejs_eb_widget.install
--- a/modules/eb_widget/dropzonejs_eb_widget.install	(revision 4354e09391062d73397bac1d3fb7c234b043694b)
+++ b/modules/eb_widget/dropzonejs_eb_widget.install	(revision 305a7c87d68c4096cf831406319efd477f59bab5)
@@ -98,3 +98,36 @@
     }
   }
 }
+
+/**
+ * Update configurations for new chunking keys.
+ */
+function dropzonejs_eb_widget_update_8203() {
+  /** @var \Drupal\entity_browser\Entity\EntityBrowser[] $entity_browsers */
+  $entity_browsers = EntityBrowser::loadMultiple();
+
+  foreach ($entity_browsers as $entity_browser) {
+    /** @var \Drupal\entity_browser\WidgetInterface[] $widgets */
+    $widgets = $entity_browser->getWidgets();
+
+    $has_config_changes = FALSE;
+    foreach ($widgets as $widget) {
+      if ($widget instanceof DropzoneJsEbWidget) {
+        $config = $widget->getConfiguration();
+
+        $config['settings']['chunking'] = FALSE;
+        $config['settings']['force_chunking'] = FALSE;
+        $config['settings']['chunk_size'] = 2000000;
+        $config['settings']['parallel_chunk_uploads'] = FALSE;
+        $config['settings']['retry_chunks'] = FALSE;
+        $config['settings']['retry_chunks_limit'] = 3;
+        $widget->setConfiguration($config);
+        $has_config_changes = TRUE;
+      }
+    }
+
+    if ($has_config_changes) {
+      $entity_browser->save();
+    }
+  }
+}
Index: modules/eb_widget/src/Plugin/EntityBrowser/Widget/DropzoneJsEbWidget.php
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/modules/eb_widget/src/Plugin/EntityBrowser/Widget/DropzoneJsEbWidget.php b/modules/eb_widget/src/Plugin/EntityBrowser/Widget/DropzoneJsEbWidget.php
--- a/modules/eb_widget/src/Plugin/EntityBrowser/Widget/DropzoneJsEbWidget.php	(revision 4354e09391062d73397bac1d3fb7c234b043694b)
+++ b/modules/eb_widget/src/Plugin/EntityBrowser/Widget/DropzoneJsEbWidget.php	(revision 305a7c87d68c4096cf831406319efd477f59bab5)
@@ -142,6 +142,13 @@
       'resize_quality' => 1,
       'resize_method' => 'contain',
       'thumbnail_method' => 'contain',
+      'min_resolution' => NULL,
+      'chunking' => FALSE,
+      'force_chunking' => FALSE,
+      'chunk_size' => 2000000,
+      'parallel_chunk_uploads' => FALSE,
+      'retry_chunks' => FALSE,
+      'retry_chunks_limit' => 3,
     ] + parent::defaultConfiguration();
   }
 
@@ -166,6 +173,9 @@
       '#extensions' => $config['settings']['extensions'],
       '#max_files' => ($cardinality > 0) ? $cardinality : 0,
       '#clientside_resize' => $config['settings']['clientside_resize'],
+      '#min_resolution_x' => $min_resolution[0],
+      '#min_resolution_y' => $min_resolution[1],
+      '#chunking' => $config['settings']['chunking'],
     ];
 
     if ($config['settings']['clientside_resize']) {
@@ -176,6 +186,14 @@
       $form['upload']['#thumbnail_method'] = $config['settings']['thumbnail_method'];
     }
 
+    if ($config['settings']['chunking']) {
+      $form['upload']['#force_chunking'] = $config['settings']['force_chunking'];
+      $form['upload']['#chunk_size'] = $config['settings']['chunk_size'];
+      $form['upload']['#parallel_chunk_uploads'] = $config['settings']['parallel_chunk_uploads'];
+      $form['upload']['#retry_chunks'] = $config['settings']['retry_chunks'];
+      $form['upload']['#retry_chunks_limit'] = $config['settings']['retry_chunks_limit'];
+    }
+
     $form['#attached']['library'][] = 'dropzonejs/widget';
     // Disable the submit button until the upload sucesfully completed.
     $form['#attached']['library'][] = 'dropzonejs_eb_widget/common';
@@ -409,7 +427,6 @@
       '#default_value' => $configuration['extensions'],
     ];
 
-
     $exif_found = $this->libraryDiscovery->getLibraryByName('dropzonejs', 'exif-js');
 
     $form['clientside_resize'] = [
@@ -508,6 +525,78 @@
           ],
         ]
       ]
+    ];
+
+    $form['chunking'] = [
+      '#type' => 'checkbox',
+      '#title' => $this->t('Enable chunking'),
+      '#default_value' => $configuration['chunking'],
+    ];
+
+    $form['force_chunking'] = [
+      '#type' => 'checkbox',
+      '#title' => $this->t('Force chunking'),
+      '#default_value' => $configuration['force_chunking'],
+      '#states' => [
+        'visible' => [
+          ':input[name="table[' . $this->uuid() .  '][form][chunking]"]' => [
+            'checked' => TRUE,
+          ],
+        ],
+      ],
+    ];
+
+    $form['chunk_size'] = [
+      '#type' => 'number',
+      '#title' => $this->t('Chunk size'),
+      '#default_value' => $configuration['chunk_size'],
+      '#min' => 1,
+      '#states' => [
+        'visible' => [
+          ':input[name="table[' . $this->uuid() .  '][form][chunking]"]' => [
+            'checked' => TRUE,
+          ],
+        ],
+      ],
+    ];
+
+    $form['parallel_chunk_uploads'] = [
+      '#type' => 'checkbox',
+      '#title' => $this->t('Parallel chunk uploads'),
+      '#default_value' => $configuration['parallel_chunk_uploads'],
+      '#states' => [
+        'visible' => [
+          ':input[name="table[' . $this->uuid() .  '][form][chunking]"]' => [
+            'checked' => TRUE,
+          ],
+        ],
+      ],
+    ];
+
+    $form['retry_chunks'] = [
+      '#type' => 'checkbox',
+      '#title' => $this->t('Retry chunks'),
+      '#default_value' => $configuration['retry_chunks'],
+      '#states' => [
+        'visible' => [
+          ':input[name="table[' . $this->uuid() .  '][form][chunking]"]' => [
+            'checked' => TRUE,
+          ],
+        ],
+      ],
+    ];
+
+    $form['retry_chunks_limit'] = [
+      '#type' => 'number',
+      '#title' => $this->t('Retry chunks limit'),
+      '#default_value' => $configuration['retry_chunks_limit'],
+      '#states' => [
+        'visible' => [
+          ':input[name="table[' . $this->uuid() .  '][form][chunking]"]' => [
+            'checked' => TRUE,
+          ],
+        ],
+      ],
     ];
 
     return $form;
Index: src/Element/DropzoneJs.php
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/Element/DropzoneJs.php b/src/Element/DropzoneJs.php
--- a/src/Element/DropzoneJs.php	(revision 4354e09391062d73397bac1d3fb7c234b043694b)
+++ b/src/Element/DropzoneJs.php	(revision 305a7c87d68c4096cf831406319efd477f59bab5)
@@ -33,6 +33,9 @@
  * - #clientside_resize (bool)
  *   Whether or not to use DropzoneJS clientside resizing. It requires v4.4.0+
  *   version of the library.
+ * - #chunking (bool)
+ *   Whether or not to use DropzoneJS chunking. It requires v5.7.0+
+ *   version of the library.
  *
  * Optional options are:
  * - #resize_width (integer)
@@ -48,6 +51,24 @@
  * - #thumbnail_method (string).
  *   (optional) Accepts 'contain', which scales the image, or 'crop' which crops
  *   the image. Defaults to 'contain'.
+ * - #show_alt (boolean).
+ *   (optional) TRUE if alt text input should be shown.
+ * - #show_title (boolean).
+ *   (optional) TRUE if title text input should be shown.
+ * - #force_chunking (bool).
+ *   (optional) Whether or not to force chunking even if the file size is below
+ *   chunk_size. Defaults to false.
+ * - #chunk_size (integer).
+ *   (optional) The chunk size in bytes. Defaults to 2000000.
+ * - #parallel_chunk_uploads (bool).
+ *   (optional) Whether or not individual chunks of a file can be uplaoded
+ *   in parallel. Defaults to false.
+ * - #retry_chunks (bool).
+ *   (optional) Whether or not a chunk should be retried if it fails.
+ *   Defaults to false.
+ * - #retry_chunks_limit (integer).
+ *   (optional) The number of times a chunk should be retried.
+ *   Defaults to 3.
  *
  * @todo Not sure about the version for clientside.
  *
@@ -90,11 +111,22 @@
       '#type' => 'hidden',
       // @todo Handle defaults.
       '#default_value' => '',
-      // If we send a url with a token through drupalSettings the placeholder
-      // doesn't get replaced, because the actual scripts markup is not there
-      // yet. So we pass this information through a data attribute.
-      '#attributes' => ['data-upload-path' => Url::fromRoute('dropzonejs.upload')->toString()],
-    ];
+    ];
+
+    // If we send a url with a token through drupalSettings the placeholder
+    // doesn't get replaced, because the actual scripts markup is not there
+    // yet. So we pass this information through a data attribute.
+    if (!empty($element['#chunking'])) {
+      $element['uploaded_files']['#attributes'] = [
+        'data-upload-path' => Url::fromRoute('dropzonejs.chunked.upload')->toString(),
+        'data-finalize-path' => Url::fromRoute('dropzonejs.chunked.finalize')->toString(),
+      ];
+    }
+    else {
+      $element['uploaded_files']['#attributes'] = [
+        'data-upload-path' => Url::fromRoute('dropzonejs.upload')->toString(),
+      ];
+    }
 
     if (empty($element['#max_filesize'])) {
       $element['#max_filesize'] = Environment::getUploadMaxSize();
@@ -154,6 +186,17 @@
       array_unshift($element['#attached']['library'], 'dropzonejs/exif-js');
     }
 
+    if (!empty($element['#chunking'])) {
+      $element['#attached']['drupalSettings']['dropzonejs']['instances'][$element['#id']] += [
+        'chunking' => $element['#chunking'],
+        'forceChunking' => !empty($element['#force_chunking']) ? $element['#force_chunking'] : FALSE,
+        'chunkSize' => !empty($element['#chunk_size']) ? $element['#chunk_size'] : 2000000,
+        'parallelChunkUploads' => !empty($element['#parallel_chunk_uploads']) ? $element['#parallel_chunk_uploads'] : FALSE,
+        'retryChunks' => !empty($element['#retry_chunks']) ? $element['#retry_chunks'] : FALSE,
+        'retryChunksLimit' => !empty($element['#retry_chunks_limit']) ? $element['#retry_chunks_limit'] : 3,
+      ];
+    }
+
     static::setAttributes($element, ['dropzone-enable']);
     return $element;
   }
Index: src/UploadException.php
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/UploadException.php b/src/UploadException.php
--- a/src/UploadException.php	(revision 4354e09391062d73397bac1d3fb7c234b043694b)
+++ b/src/UploadException.php	(revision 305a7c87d68c4096cf831406319efd477f59bab5)
@@ -39,6 +39,11 @@
    */
   const FILE_UPLOAD_ERROR = 106;
 
+  /**
+   * Chunks assembly resulted in error.
+   */
+  const ASSEMBLE_CHUNKS_ERROR = 107;
+
   /**
    * Code to error message mapping.
    *
@@ -51,6 +56,7 @@
     self::DESTINATION_FOLDER_ERROR => 'Failed to open temporary directory for write.',
     self::FILENAME_ERROR => 'Invalid temporary file name.',
     self::FILE_UPLOAD_ERROR => 'The file upload resulted in an error on php level. See http://php.net/manual/en/features.file-upload.errors.php',
+    self::ASSEMBLE_CHUNKS_ERROR => 'Failed to assemble chunks.',
   ];
 
   /**
Index: src/UploadHandler.php
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/UploadHandler.php b/src/UploadHandler.php
--- a/src/UploadHandler.php	(revision 4354e09391062d73397bac1d3fb7c234b043694b)
+++ b/src/UploadHandler.php	(revision 305a7c87d68c4096cf831406319efd477f59bab5)
@@ -72,9 +72,7 @@
   /**
    * {@inheritdoc}
    */
-  public function getFilename(UploadedFile $file) {
-    $original_name = $file->getClientOriginalName();
-
+  public function getFilename($original_name) {
     // There should be a filename and it should not contain a semicolon,
     // which we use to separate filenames.
     if (!isset($original_name)) {
@@ -138,7 +136,7 @@
     }
 
     // Open temp file.
-    $tmp = $this->dropzoneSettings->get('tmp_upload_scheme') . '://' . $this->getFilename($file);
+    $tmp = $this->dropzoneSettings->get('tmp_upload_scheme') . '://' . $this->getFilename($file->getClientOriginalName());
     if (!($out = fopen($tmp, $this->request->request->get('chunk', 0) ? 'ab' : 'wb'))) {
       throw new UploadException(UploadException::OUTPUT_ERROR);
     }
Index: src/UploadHandlerInterface.php
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/UploadHandlerInterface.php b/src/UploadHandlerInterface.php
--- a/src/UploadHandlerInterface.php	(revision 4354e09391062d73397bac1d3fb7c234b043694b)
+++ b/src/UploadHandlerInterface.php	(revision 305a7c87d68c4096cf831406319efd477f59bab5)
@@ -10,17 +10,17 @@
 interface UploadHandlerInterface {
 
   /**
-   * Reads, checks and return filename of a file being uploaded.
+   * Sanitizes the filename of the uploaded file.
    *
-   * @param \Symfony\Component\HttpFoundation\File\UploadedFile $file
-   *   An instance of UploadedFile.
+   * @param string $original_name
+   *   The original filename.
    *
    * @return string
    *   The sanitized filename.
    *
    * @throws \Drupal\dropzonejs\UploadException
    */
-  public function getFilename(UploadedFile $file);
+  public function getFilename($original_name);
 
   /**
    * Handles an uploaded file.
Index: tests/modules/dropzonejs_test/config/install/entity_browser.browser.dropzonejs_eb_test.yml
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/tests/modules/dropzonejs_test/config/install/entity_browser.browser.dropzonejs_eb_test.yml b/tests/modules/dropzonejs_test/config/install/entity_browser.browser.dropzonejs_eb_test.yml
--- a/tests/modules/dropzonejs_test/config/install/entity_browser.browser.dropzonejs_eb_test.yml	(revision 4354e09391062d73397bac1d3fb7c234b043694b)
+++ b/tests/modules/dropzonejs_test/config/install/entity_browser.browser.dropzonejs_eb_test.yml	(revision 305a7c87d68c4096cf831406319efd477f59bab5)
@@ -32,6 +32,12 @@
       resize_quality: !!float 1
       resize_method: contain
       thumbnail_method: contain
+      chunking: false
+      force_chunking: false
+      chunk_size: 2000000
+      parallel_chunk_uploads: false
+      retry_chunks: false
+      retry_chunks_limit: 3
       submit_text: 'Select entities'
       auto_select: false
     uuid: 44b1e6ea-637d-4dd6-b79e-edeefc546c1c
